### PR TITLE
declaration: allow store paths

### PR DESCRIPTION
**Pull request by MarcelCoding**

### CODE REVIEW

File: ixx/src/action/index.rs [hunk 1]
```diff
@@ -53,52 +53,82 @@ pub(crate) fn index(module: IndexModule) -> anyhow::Result<()> {
...
+  fn test_update_declaration() {
+    assert_eq!(
+      update_declaration(
+        &Url::parse("https://nuschtos.de").unwrap(),
+        Declaration::StorePath(
+          "/nix/store/vgvk6q3zsjgb66f8s5cm8djz6nmcag1i-source/modules/initrd.nix".to_string()
+        )
+      )
+      .unwrap(),
+      Url::parse("https://nuschtos.de/modules/initrd.nix").unwrap()
```
Comment (SuperSandro2000) [inline ixx/src/action/index.rs:119]: We should use example.com

File: ixx/src/action/index.rs [hunk 2]
```diff
@@ -53,52 +53,82 @@ pub(crate) fn index(module: IndexModule) -> anyhow::Result<()> {
...
+
+  use crate::{action::index::update_declaration, option::Declaration};
+
+  #[test]
+  fn test_update_declaration() {
+    assert_eq!(
+      update_declaration(
+        &Url::parse("https://nuschtos.de").unwrap(),
+        Declaration::StorePath(
+          "/nix/store/vgvk6q3zsjgb66f8s5cm8djz6nmcag1i-source/modules/initrd.nix".to_string()
```
Comment (SuperSandro2000) [inline ixx/src/action/index.rs:115]:
```suggestion
          "/nix/store/vgvk6q3zsjgb66f8s5cm8djz6nmcag1i-source-two/modules/initrd.nix".to_string()
```

that doesn't work with the 51 chars hardcoded from above?

Comment (MarcelCoding) [reply ixx/src/action/index.rs:115]: Wasn't it always just "source"?

Comment (SuperSandro2000) [reply ixx/src/action/index.rs:131]: No, that's the name of the derivation which is by default source but could be anything except / IIRC

**Pull request merged**